packages/browser-agent-core/common/window/load.js

```diff
@@ -0,0 +1,23 @@
+import { eventListenerOpts } from '../event-listener/event-listener-opts'
+
+const win = window
+const doc = win.document
+const ADD_EVENT_LISTENER = 'addEventListener'
+const ATTACH_EVENT = 'attachEvent'
+
+function checkState (cb) {
+  if (doc.readyState === 'complete') cb()
+}
+
+export function onWindowLoad(cb) {
+  checkState(cb)
```

Comment:
This checkState will call the onWindowLoaded callback if the agent loads after the window.load and set the onload and load timings incorrectly.
https://github.com/newrelic/newrelic-browser-agent/blob/7e13aea02cc7ac928592bfaf71e5a6f0d18bac23/packages/browser-agent-core/features/page-view-event/instrument/index.js#L18-L28

Reply:
The old agent is clearer that `measureWindowLoaded()` (`windowLoaded()` in the old agent) should only ever be called more than once:

https://github.com/newrelic/newrelic-browser-agent/blob/7e13aea02cc7ac928592bfaf71e5a6f0d18bac23/loader/index.js#L77-L79

Reply:
To be clear, the functions you linked, `onWindowLoad()` and `onDOMContentLoaded()`, are just event listeners that early resolve if the event has already happened.  The behavior you are referring to (setting timings, now(), mark, etc), should be handled by the callers (in the page-view-event module, for instance) and not in these functions, as these are generic.  What these functions could do however, is return a flag that lets the caller know that it resolved because it had already triggered, letting the callers know not to implement timings or whatnot.

Reply:
(its also used by the CDN to inject the aggregator files)